pkg/logs/decoder/decoder.go

```diff
@@ -74,6 +75,7 @@ type Decoder struct {
 	lineParser      LineParser
 	contentLenLimit int
 	rawDataLen      int
+	linesDecoded    int64
```

Comment:
might be worth mentioning in a comment that this tracks the number of raw lines decoded _before_ they go through the parser (which may re-assemble lines together, depending on the source) and the handler (which may re-assemble lines together, depending on multi-line rules)